### ISSUE TITLE
Calamares renderes Black on Gnome 43.1 when using Administrator Mode (polkit/sudo)

### ISSUE THREAD

**Issue by philmmanjaro** [calamares/calamares#2074]
**Describe the bug**
It seems that Gnome 43.1 has some issue to render Calamares properly when using polkit or sudo. It simply displays the content completely black. Keyboard navigation works and it also displays dialogs fine if you X it.

**To Reproduce**
1. Get a [daily Gnome build from Manjaro](https://github.com/manjaro-gnome/download/releases/tag/202211130342)
2. Boot the image up in a Virtual Machine
3. Start Calamares with Admin rights: result is black rendering
4. Start Calamares with normal User rights: Calamares renders properly

**Expected behavior**
Render Calamares properly. On Wayland session a `QT_QPA_PLATFORM=wayland` might render it properly even with Admin rights

**Screenshots and Logs**
![image](https://user-images.githubusercontent.com/5050022/201517194-fb955f80-17a5-4141-b5a8-54c2d255e335.png)
![image](https://user-images.githubusercontent.com/5050022/201517200-b661bd92-f090-475c-ac88-9fbaa8314786.png)

**Additional context**
Calamares Version tested: 3.2.61-6

**Comment by philmmanjaro** [comment]
Seems to be some **mesa** issue with qxl. Running 22.1.7 rendered it just fine

![image](https://user-images.githubusercontent.com/5050022/201518392-455a6ec6-20aa-4373-b14e-fd46a12d75f6.png)
***on Virtualbox 7***

**Comment by philmmanjaro** [comment]
**Mesa** 22.2.0 is also "broken" in that regard. So a `git-bisect` of mesa seems to be needed to find out. **mesa-git** seems to have the same issues reported.

**Comment by philmmanjaro** [comment]
I've tested now an [ISO with Gnome 42.5](https://github.com/manjaro-gnome/download/releases/tag/202210310353) and **mesa 22.2.1**. Same black screen when using polkit/sudo

**Comment by ptr1337** [comment]
> I've tested now an [ISO with Gnome 42.5](https://github.com/manjaro-gnome/download/releases/tag/202210310353) and **mesa 22.2.1**. Same black screen when using polkit/sudo

We are running mesa 22.2.3 and it has the same issue.

**Comment by philmmanjaro** [comment]
It all points out to `git-bisect` mesa to see what it is. Anyway it is niche problem after all. Regular Qt5 apps you can start with admin privileges ...

**Comment by oberon-manjaro** [comment]
Just did the git bisect and pinned down the offending commit:
https://gitlab.freedesktop.org/mesa/mesa/-/commit/68e89401140d1b3a17052899c54577de3f008b67

Building latest version 22.2.4 of mesa with just this commit reverted results in calamares_polkit working properly again.

```
diff -rupN a/src/glx/drisw_glx.c b/src/glx/drisw_glx.c
--- a/src/glx/drisw_glx.c	2022-11-17 00:02:11.000000000 +0100
+++ b/src/glx/drisw_glx.c	2022-11-22 16:57:14.592161043 +0100
@@ -884,16 +884,9 @@ check_xshm(Display *dpy)
    xcb_void_cookie_t cookie;
    xcb_generic_error_t *error;
    int ret = True;
-   xcb_query_extension_cookie_t shm_cookie;
-   xcb_query_extension_reply_t *shm_reply;
-   bool has_mit_shm;
+   int ignore;

-   shm_cookie = xcb_query_extension(c, 7, "MIT-SHM");
-   shm_reply = xcb_query_extension_reply(c, shm_cookie, NULL);
-
-   has_mit_shm = shm_reply->present;
-   free(shm_reply);
-   if (!has_mit_shm)
+   if (!XQueryExtension(dpy, "MIT-SHM", &xshm_opcode, &ignore, &ignore))
       return False;

    cookie = xcb_shm_detach_checked(c, 0);
```

**Comment by philmmanjaro** [comment]
Ya can confirm that on my end too. Lets see what Mesa Devs might say.

**Comment by oberon-manjaro** [comment]
git bisect log for reference
[mesa_git-bisect.log](https://github.com/calamares/calamares/files/10070057/mesa_git-bisect.log)

**Comment by demmm** [comment]
Since this is now clearly not a Calamares issue, but as pointed out, issue is with mesa, and links to needed options/fixes are posted, time to close the issue here.

**Comment by philmmanjaro** [comment]
Seems to been fixed via https://gitlab.freedesktop.org/mesa/mesa/-/commit/57b7102ef96a282128ead4e7a8c03c7fdb53e71a which was also added to 22.3.1 of **mesa**. @demmm confirmed it from her side too.

**Comment by yochananmarqos** [comment]
> Seems to been fixed

@philmmanjaro Well, then please close the issue. :wink:

**Comment by demmm** [comment]
@yochananmarqos This has been closed since Nov 22
